p2p/net/network.go

```diff
@@ -304,16 +307,21 @@ func (n *Net) accept(listen net.Listener) {
 		n.tcpSocketConfig(conn) // TODO maybe only set this after session handshake to prevent denial of service with big messages
 		c := newConnection(netConn, n, nil, nil, n.config.MsgSizeLimit, n.config.ResponseTimeout, n.logger)
 		go func(con Connection) {
-			defer func() { pending <- struct{}{} }()
-			err := c.setupIncoming(n.config.SessionTimeout)
-			if err != nil {
+			if err := c.setupIncoming(n.config.SessionTimeout); err != nil {
 				n.logger.Event().Warning("conn_incoming_failed", log.String("remote", c.remoteAddr.String()), log.Err(err))
 				return
 			}
-			go c.beginEventProcessing()
+
+			// Asynchronously handle the new connection
+			func() {
```

Comment:
I really don't understand, you meant to run this in a goroutine?

Reply:
Yes. Otherwise, the `defer` runs immediately when the `func` ends, which happens immediately, since `go c.beginEventProcessing()` is async. That's not what we want. We want to defer the `pending <- struct{}{}` until `c.beginEventProcessing()` returns.

Reply:
Yes, typo, will fix! Thanks